### PR TITLE
unit-tests-jest exercise - Cynthia Huynh

### PR DESCRIPTION
**Pull request by HuynhCynthia**

### CODE REVIEW

File: unit-tests-jest/numbers.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  return numbers.filter((n) => !(n % 2));
+}
+
+/**
+ * Returns a number formatted in dollars and cents.
+ */
+export function toDollars(amount) {
+  if (!Number.isFinite(amount)) {
+    return;
+  } else if (!Number.isInteger(amount)) {
```
Comment (RobertGardner) [inline unit-tests-jest/numbers.js:14]: I had intended this function to work for non-integer values, so `2.15` formats as `$2.15`.

File: unit-tests-jest/numbers.test.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
+import { evenNumbers, toDollars, divideBy, multiplyBy } from './numbers';
+
+describe('evenNumbers', () => {
+  it('returns the even numbers', () => {
+    const numbers = [1, 4, 5, 10, 0];
+    const result = evenNumbers(numbers);
+    expect(result).toEqual([4, 10, 0]);
+  });
```
Comment (RobertGardner) [inline unit-tests-jest/numbers.test.js:8]: There are other edge cases you could check here, such as a empty array, non-integers.

File: unit-tests-jest/numbers.test.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    expect(result).toEqual(undefined);
+    const amountNeg = -6;
+    const resultNeg = toDollars(amountNeg);
+    expect(resultNeg).toEqual(undefined);
+    const amountFloat = 3.3;
+    const resultFloat = toDollars(amountFloat);
+    expect(resultFloat).toEqual(undefined);
+    const resultNaN = toDollars(NaN);
+    expect(resultNaN).toEqual(undefined);
+  });
```
Comment (RobertGardner) [inline unit-tests-jest/numbers.test.js:29]: At least you tested what you implemented, even if it was different than what I had intended! That is very good and what you should do.

File: unit-tests-jest/numbers.test.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+});
+
+describe('toDollars', () => {
+  it('returns int dollar amount', () => {
+    const amount = 3;
+    const result = toDollars(amount);
+    expect(result).toEqual('$3.00');
+  });
+  it('only allows positive int', () => {
```
Comment (RobertGardner) [inline unit-tests-jest/numbers.test.js:17]: I would break this into several tests instead of doing it all in one test. To do that you have to write more test code, but it will make debugging easier.

File: unit-tests-jest/numbers.test.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    const result = divideBy(numbers, divisor);
+    expect(result).toEqual([0, 1, 2, 3]);
+  });
+  it('Does not modify the original array', () => {
+    const numbers = [1, 3, 5];
+    const divisor = 2;
+    divideBy(numbers, divisor);
+    expect(numbers).toEqual([1, 3, 5]);
+  });
+  it('divisor and array values must be type number', () => {
```
Comment (RobertGardner) [inline unit-tests-jest/numbers.test.js:45]: Should probably be 2 separate tests.

File: unit-tests-jest/numbers.test.js [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+});
+
+describe('multiplyBy', () => {
+  it('returns modified object by multiplying the value of each key', () => {
+    const obj = { red: 2, blue: 4 };
+    const multiplier = 2;
+    const result = multiplyBy(obj, multiplier);
+    expect(result).toEqual({ blue: 8, red: 4 });
+  });
+  it('multiplier and object values must be type number', () => {
```
Comment (RobertGardner) [inline unit-tests-jest/numbers.test.js:64]: And this one should be 2 tests, perhaps more.

**Review by RobertGardner — APPROVED** [review]
You did a good job here. I made some comments that would improve the test quality, but you tested what you implemented, which is great!
